src/cms/models/users/user.py

```diff
@@ -0,0 +1,240 @@
+        help_text=_("The date and time when the user did read the chat the last time"),
+    )
+    expert_mode = models.BooleanField(
+        default=False,
+        verbose_name=_("experienced user"),
+        help_text=_(
+            "Enable this option to show up additional features like XLIFF import/export, page filtering, mirrored pages, page-based permissions, Do-Not-Translate-Tag and recurring events"
+        ),
+    )
+
+    @property
+    def role(self):
+        """
+        We refer to Django user groups as roles.
+
+        :return: The role of this user
+        :rtype: ~cms.models.users.role.Role
+        """
+        if self.groups.exists():
+            return self.groups.first().role
+        return None
+
+    @property
+    def full_user_name(self):
+        """
+        Return the full name of the user. If either the first or the last name are present, return them, otherwise
+        return the username.
+
+        :return: The full name of the user
+        :rtype: str
+        """
+        return self.get_full_name() or self.get_username()
+
+    @property
+    def unread_chat_messages(self):
+        """
+        Return all unread messages of this user
+
+        :return: The unread messages of this user
+        :rtype: ~django.db.models.query.QuerySet [ ~cms.models.chat.chat_message.ChatMessage ]
+        """
+        return ChatMessage.history.filter(sent_datetime__gt=self.chat_last_visited)
+
+    def update_chat_last_visited(self):
+        """
+        Update the :attr:`~cms.models.users.user.User.chat_last_visited` to the current time
+
+        :return: Return the previous :attr:`~cms.models.users.user.User.chat_last_visited` value
+        :rtype: ~datetime.datetime
+        """
+        previous_chat_last_visited = self.chat_last_visited
+        self.chat_last_visited = timezone.now()
+        self.save()
+        logger.debug(
+            "Field chat_last_visited of %r updated from %s to %s",
+            self,
+            previous_chat_last_visited.strftime("%Y-%m-%d %H:%M:%S"),
+            self.chat_last_visited.strftime("%Y-%m-%d %H:%M:%S"),
+        )
+        return previous_chat_last_visited
+
+    def __str__(self):
+        """
+        This overwrites the default Django :meth:`~django.db.models.Model.__str__` method which would return ``User object (id)``.
+        It is used in the Django admin backend and as label for ModelChoiceFields.
+
+        :return: A readable string representation of the user
+        :rtype: str
+        """
+        return self.full_user_name
+
+    def __repr__(self):
+        """
+        This overwrites the default Django ``__repr__()`` method which would return ``<User: User object (id)>``.
+        It is used for logging.
+
+        :return: The canonical string representation of the user
+        :rtype: str
+        """
+        optional_fields = ""
+        if not self._state.adding:
+            if self.is_staff:  # pylint: disable=W0125
```

Comment:
```suggestion
            if self.is_staff:
```
I'd propose to use the string identifiers of pylint warning disable statements...
In this case, it's `# pylint: disable=using-constant-test`, which would make it more clear that it's a remnant of a debugging test like `if False` or similar and can be removed in the production code.